cub/device/device_run_length_encode.cuh

```diff
@@ -28,245 +27,366 @@
  */
 
 #pragma once
 
-#include <stdio.h>
 #include <iterator>
+#include <stdio.h>
 
-#include "../config.cuh"
-#include "dispatch/dispatch_rle.cuh"
-#include "dispatch/dispatch_reduce_by_key.cuh"
+#include <cub/config.cuh>
+#include <cub/device/dispatch/dispatch_reduce_by_key.cuh>
+#include <cub/device/dispatch/dispatch_rle.cuh>
 
 CUB_NAMESPACE_BEGIN
 
 
 /**
- * \brief DeviceRunLengthEncode provides device-wide, parallel operations for demarcating "runs" of same-valued items within a sequence residing within device-accessible memory. ![](run_length_encode_logo.png)
- * \ingroup SingleModule
+ * @brief DeviceRunLengthEncode provides device-wide, parallel operations for 
+ *        demarcating "runs" of same-valued items within a sequence residing 
+ *        within device-accessible memory. ![](run_length_encode_logo.png)
+ * @ingroup SingleModule
  *
- * \par Overview
- * A <a href="http://en.wikipedia.org/wiki/Run-length_encoding"><em>run-length encoding</em></a>
- * computes a simple compressed representation of a sequence of input elements such that each
- * maximal "run" of consecutive same-valued data items is encoded as a single data value along with a
- * count of the elements in that run.
+ * @par Overview
+ * A <a href="http://en.wikipedia.org/wiki/Run-length_encoding">*run-length encoding*</a>
+ * computes a simple compressed representation of a sequence of input elements 
+ * such that each maximal "run" of consecutive same-valued data items is 
+ * encoded as a single data value along with a count of the elements in that 
+ * run.
  *
- * \par Usage Considerations
- * \cdp_class{DeviceRunLengthEncode}
+ * @par Usage Considerations
+ * @cdp_class{DeviceRunLengthEncode}
  *
- * \par Performance
- * \linear_performance{run-length encode}
+ * @par Performance
+ * @linear_performance{run-length encode}
  *
- * \par
- * The following chart illustrates DeviceRunLengthEncode::RunLengthEncode performance across
- * different CUDA architectures for \p int32 items.
- * Segments have lengths uniformly sampled from [1,1000].
+ * @par
+ * The following chart illustrates DeviceRunLengthEncode::RunLengthEncode 
+ * performance across different CUDA architectures for `int32` items.
+ * Segments have lengths uniformly sampled from `[1, 1000]`.
  *
- * \image html rle_int32_len_500.png
- *
- * \par
- * \plots_below
+ * @image html rle_int32_len_500.png
  *
+ * @par
+ * @plots_below
  */
 struct DeviceRunLengthEncode
 {
+  /**
+   * @brief Computes a run-length encoding of the sequence \p d_in.
+   *
+   * @par
+   * - For the *i*<sup>th</sup> run encountered, the first key of the run and 
+   *   its length are written to `d_unique_out[i]` and `d_counts_out[i]`,
+   *   respectively.
+   * - The total number of runs encountered is written to `d_num_runs_out`.
+   * - The `==` equality operator is used to determine whether values are 
+   *   equivalent
+   * - Let `out` represent a set of following ranges 
```

Comment:
This seems equivalent to just stating that there shall be no overlap between any of the 4 listed ranges, with no need to group the first three together as `out`. Is there some nuance I'm missing here?

Reply:
There's no technical nuance you are missing. This phrasing just represented my mental model of these iterators :smile: The suggested phrasing might be easier to read, so I've updated the docs, thanks!